rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -44,40 +47,38 @@ def train(
                 entity_val = example.get(TEXT)[entity["start"] : entity["end"]]
                 self.add_entities_if_synonyms(entity_val, str(entity.get("value")))
 
-    def process(self, message: Message, **kwargs: Any) -> None:
+        return self.persist()
+
+    def process(self, message: Optional[Message], **kwargs: Any) -> Optional[Message]:
```

Comment:
I want to make all the `process` methods handle a list of messages as part of deriving prediction graph. Then we can get rid of `process_training_data` data everywhere

Reply:
that sounds awesome!

Reply:
There seems to sometimes be slight differences between the training process and the normal process, so will keep `process_training_data` for now. Although I have still made process handle lists of messages which makes the nlu e2e prediction pipeline more consistent with the rest of the predict graph.